packages/frontend-v2/pages/home.tsx

```diff
@@ -0,0 +1,128 @@
+import { NextPage } from "next";
+import { Plan, ScheduleCourse } from "../components";
+import {
+  fetchStudentAndPrepareForDnd,
+  useStudentWithPlans,
+} from "../hooks/useStudentWithPlans";
+import {
+  DndContext,
+  DragEndEvent,
+  DragOverlay,
+  DragStartEvent,
+} from "@dnd-kit/core";
+import {
+  cleanDndIdsFromPlan,
+  findCourseByDndId,
+  logger,
+  toast,
+  updatePlanForStudent,
+  updatePlanOnDragEnd,
+} from "../utils";
+import { API } from "@graduate/api-client";
+import { PlanModel, ScheduleCourse2 } from "@graduate/common";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import { Button } from "@chakra-ui/react";
+
+const HomePage: NextPage = () => {
+  const router = useRouter();
+  const { error, student, mutateStudent } = useStudentWithPlans();
+
+  // keep track of the course being dragged around so that we can display the drag overlay
+  const [activeCourse, setActiveCourse] =
+    useState<ScheduleCourse2<string> | null>(null);
+
+  if (error) {
+    logger.error("HomePage", error);
+    return <p>Error fetching student</p>;
+  }
+
+  if (!student) {
+    return <p>Loading...</p>;
+  }
+
+  const primaryPlan = student.plans.find((p) => student.primaryPlanId === p.id);
+
+  if (!primaryPlan) {
+    // TODO: Handle no plan/no primary plan case
+    throw new Error("Student doesn't have a valid primary plan set");
+  }
+
+  const handleDragStart = (event: DragStartEvent): void => {
+    // find the course that is being dragged and set it as the active course
+    const { active } = event;
+    const activeCourse = findCourseByDndId(
+      primaryPlan.schedule,
+      active.id as string // dnd ids are strings
```

Comment:
If im understanding correctly, this type has to match the generic for a PlanModel<T>. If this is the case, should make a note that this type must match that generic in case we want to change the type of the id for whatever reason

Reply:
AHH good point!!